### ISSUE TITLE
Stale acu position data in hwp supervisor during `check_acu_ok_for_spinup()`

### ISSUE THREAD

**Issue by BrianJKoopman** [simonsobs/socs#799]
Related to this issue, satp3 had similar problem on Nov 29th 2024, by following schedule https://site.simonsobs.org/satp3/nextline/db/runs/2202

```
run.wait_until('2024-11-29T18:44:23.980000+00:00')
run.acu.move_to(az=180.0, el=40.0)
run.acu.move_to(az=180.0, el=48.0)
######## HWP spinning up ##########
run.hwp.set_freq(freq=2) # start HWP rotation
```

In this case, I think the safety was triggered due to the time lag of acu monitoring.
hwp spin up was called right after the telescope has moved from 40 to 48, and the hwp_supervisor checked the acu position that was updated slightly before, and it was lower than 48.
One of the solutions is to just add a time buffer in the schedule.

_Originally posted by @ykyohei in https://github.com/simonsobs/socs/issues/788#issuecomment-2508417398_

As discussed in that original thread, it seems most likely the HWP supervisor has an outdated cache of the ACU elevation. Can we force a `hwp_state.acu.update()`  at the start of [`check_acu_ok_for_spinup()`](https://github.com/simonsobs/socs/blob/38566db3df2458fb864a6466713fe60ab769d045/socs/agents/hwp_supervisor/agent.py#L966)?

**Comment by jlashner** [comment]
Yes, I think forcing a state update before a check like this is reasonable. I can implement that.
